webapp/src/shared/components/Game/Game.tsx

```diff
@@ -1,72 +1,73 @@
 import * as React from 'react';
 
+import socket from '../../socket';
 import { CANVAS_WIDTH, CANVAS_HEIGHT } from './constants';
-import { GameBall, GamePaddle } from './types';
-import { initialBallState, initialPaddleState, isLose, isWin } from './physics';
-import useGameControls from './useGameControls';
-import useGameAnimation from './useGameAnimation';
+import {
+  useGameControls,
+  useGameAnimation,
+  useClientGameEngine,
+} from './hooks';
+import { GameStateContextProvider } from './context';
 
 import './Game.css';
 
-export default function Game() {
+const GAME_SERVER_MESSAGE = 'gameServerMessage';
+
+const Game = () => {
+  const { renderFrame, deltaTimeRef } = useGameAnimation();
+  const { runGameFrame } = useClientGameEngine();
+  useGameControls();
   const canvasRef = React.useRef<HTMLCanvasElement>(null);
   const requestFrameRef = React.useRef<number | null>(null);
-  const ballRef = React.useRef<GameBall>(initialBallState());
-  const paddleRef = React.useRef<GamePaddle>(initialPaddleState());
-  const [score, setScore] = React.useState(0);
+  const [score, setScore] = React.useState<number>(0);
 
-  const { drawBall, drawPaddle, drawBrick, resetGame } = useGameAnimation(
-    ballRef,
-    paddleRef,
-  );
-  useGameControls(paddleRef);
+  const gameLoop = React.useCallback(() => {
+    const canvasContext = canvasRef.current?.getContext('2d');
+    const gameState = runGameFrame(deltaTimeRef.current);
```

Comment:
This is one of the magics, we just a game frame and get the current state which is currently decoupled from the game. This way, we can get this state from the server or the client, indepently (although it will need further reconciliation as we talked yesterday)